src/selection-recommendation/recommend.py

```diff
@@ -0,0 +1,206 @@
+        elif lec_comps[0] == "Tut":
+            tutorial_times.append(lec_time)
+            tutorial_sections.append(thing_to_append)
+        elif lec_comps[0] == "Pra":
+            lab_times.append(lec_time)
+            lab_sections.append(thing_to_append)
+        else:  # shouldn't happen
+            raise ValueError
+
+    all_possible_times = []
+    all_possible_section_combs = []
+
+    # course has both tutorial and lab sections
+    if (tutorial_times != []) and (lab_times != []):
+        all_times = list(itertools.product(lecture_times, tutorial_times, lab_times))  # all combinations of lecture/tutorial/lab
+        all_section_combs = list(itertools.product(lecture_sections, tutorial_sections, lab_sections))
+
+        for i in range(len(all_times)):
+            if (time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][1])  # check for internal time conflicts
+                    and time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][2])
+                    and time_conflicts_check.no_time_conflict(all_times[i][1], all_times[i][2])):
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course has lab sections but no tutorial sections
+    elif (tutorial_times == []) and (lab_times != []):
+        all_times = list(itertools.product(lecture_times, lab_times))  # all combinations of lecture/lab
+        all_section_combs = list(itertools.product(lecture_sections, lab_sections))
+
+        for i in range(len(all_times)):
+            if time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][1]):  # check for internal time conflicts
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course has tutorial sections but no lab sections
+    elif (tutorial_times != []) and (lab_times == []):
+        all_times = list(itertools.product(lecture_times, tutorial_times))  # all combinations of lecture/tutorial
+        all_section_combs = [list(i) for i in list(itertools.product(lecture_sections, tutorial_sections))]
+
+        for i in range(len(all_times)):
+            if time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][1]):  # check for internal time conflicts
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course only has lecture sections
+    else:
+        all_possible_times = lecture_times
+        all_possible_section_combs = lecture_sections
+
+    return(all_possible_times, all_possible_section_combs)
+
+
+def create_schedule(campus, *args):
+    """
+    campus: a string - either "St. George", "Scarborough", or "Mississauga"
+    args: strings of course codes e.g. "CSC148", "COG250"
+    Returns a tuple identical in format to get_all_possible_course_times()
```

Comment:
Maybe you can copy the format to here?

Reply:
Would you please explain what's the relation between this method and the 2d array algorithm that we are gonna use

Reply:
The process_schedule() function converts the output of create_schedule() to a 2D array right? I was thinking it would be easier and more intuitive to pass this array into some cost function instead of using the output of create_schedule() which is a bit messy. It would be pretty simple for this cost function to check for things like number of consecutive classes, frequency of breaks, etc. and invoke it to return the best schedule.

Reply:
R u gonna include the "no conflict" part in the cost function? Though it's feasible, it will make the cost function messy imo